### DISCUSSION TITLE
TextView print a portion of text in different color

### DISCUSSION THREAD

**Discussion by DefaultUser1606** [tui-cs/Terminal.Gui#3810] [Q&A]
How do I print some of the text in a different color than the rest? Can anyone provide an example? All I can find in the documentation is how to change the ColorScheme which affects the entire graphical element. I just want to make some words red to be easily seen by the user.

**Comment by BDisp** [comment]
Take a look at the `Cell` record. Basically `TextView` use the `Cell` record instead of `Rune`. Thus you just set the desired color in the text in certain regions managing the current row and current column or even with selected text to change at once. If you have a text with `Hi` then you code like this:

```cs
var tv = new TextView() { Width = 20, Height = 5 };
// Get the current line
var cells = tv.GetCurrentLine();

cell[0] = new() { Rune = rune, Attribute = new Attribute (Color.Green, Color.Black) };
cell[1] = new() { Rune = rune, Attribute = new Attribute (Color.Blue, Color.Red) };
```

**Reply by DefaultUser1606** [reply]
Thank you for pointing me in the right direction. However, Cell appears to be a V2 addition. Is there a V1 way?

**Reply by BDisp** [reply]
No, it's only supported in the v2.

**Reply by DefaultUser1606** [reply]
Do I understand correctly, that there is no option to achieve this in v1? I've seen that under the hood it uses Driver.SetAttribute, but haven't been able to get it to work so far.

**Reply by BDisp** [reply]
Through the `TextView` directly no, but with a derive from it you may achieve what you want by using the `Driver.SetAttribute` but you have to deal of a lot of work 😃

**Reply by DefaultUser1606** [reply]
Thank you for clarifications

**Comment by tznind** [comment]
Here is a simple example.  Create a new Terminal.Gui project using the [dotnet new template for v1](https://github.com/gui-cs/Terminal.Gui.templates?tab=readme-ov-file):

```
dotnet new install Terminal.gui.templates::1.0.3
dotnet new tui -n myproj
cd myproj
dotnet run
```
Now update the MyView.cs file to have the following sub view.

One gotcha is that (in v1) `OnDrawContent` only works for `View` and not `Window` so make sure to subclass the correct class.

```diff

//------------------------------------------------------------------------------

//  <auto-generated>
//      This code was generated by:
//        TerminalGuiDesigner v1.0.17.0
//      You can make changes to this file and they will not be overwritten when saving.
//  </auto-generated>
// -----------------------------------------------------------------------------
namespace myproj{
    using Terminal.Gui;
    
    
    public partial class MyView {
        
        public MyView() {
            InitializeComponent();
            button1.Clicked += () => MessageBox.Query("Hello", "Hello There!", "Ok");

+            Add(new MyCoolView()
+            {
+                Width = Dim.Fill(),
+                Height = 5,
+            });
        }        
    }

+    class MyCoolView : View
+    {
+
+        public override void OnDrawContent(Rect viewport)
+        {
+            int x = 10;
+            int y = 3;
+
+            Driver.SetAttribute(new Attribute(Color.Green, this.ColorScheme.Normal.Background));
+
+           foreach (var ch in "Behold Green!")
+           {
+                AddRune(x++, y, ch);
+            }
+       }
+    }
}

```

![image](https://github.com/user-attachments/assets/4d6a7ac5-d553-4791-b733-7267cfdb5551)
